lassen/sim.py

```diff
@@ -53,18 +53,18 @@ def alu(inst:Inst, a:Data, b:Data, d:Bit) -> (Data, Bit, Bit, Bit, Bit,
             mula, mulb = a.zext(16), b.zext(16)
             mul = mula * mulb
 
+        #Negate B and add cin if subtract
+        Cin = Bit(0)
+        if alu == ALU.Sub:
+            b = ~b
+            Cin = Bit(1)
 
         C = Bit(0)
         V = Bit(0)
-        if alu == ALU.Add:
-            res, C = a.adc(b, Bit(0))
+        if alu == ALU.Add or alu == ALU.Sub:
```

Comment:
`or` is not supported as an operator in magma (we can't overload it in Python). My guess is that the `SMTBitvector` (or any of the other hwtypes classes) do not support it either (@cdonovick can you confirm?).

Can we change this to use `(...) | (...)` instead?

Reply:
Fixed with latest commit

Reply:
@leonardt you are correct that we do not support `or` on `SMTBitVector`